### PR TITLE
DOP-1883: use canonical links for redirects && remove redirect cruft

### PR DESCRIPTION
**Pull request by schmalliso**
This is part of the CROUD SEO work to reduce our redirect chain nightmares.

- Removed symlinks since they don't do anything in nextgen anyway.
- Removed redirects that applied only to legacy versions
- Updated version definition to only include versions that are published (i.e. not on docs.mongodb.com/legacy)
- Updated redirects to apply to modern versions as needed
- Updated all redirects to use canonical URL with trailing `/`
- Collapsed redirect chain (`/tutorial/connecting/` -> `/tutorial/using-the-bi/` -> `/client-applications/`)

Note: this will break the version flipper for legacy branches (since it's getting rid of most of the redirects), so should probably be accompanied by a redeploy of said legacy branches to take the new version flipper implementation. Sophie tells me that y'all are handling that, so I leave it in your capable hands!

### CODE REVIEW

File: config/redirects [hunk 1]
```diff
@@ -1,43 +1,17 @@
...
-(v1.1-*]: bi-connector/${version}/reference/mongobischema -> ${base}/${version}/components
-[*-v1.1]: bi-connector/${version}/reference/mongosqld -> ${base}/${version}/components
-raw: bi-connector/ -> ${base}/current
-[*-v1.1]: bi-connector/${version}/tutorial/connecting -> ${base}/${version}/
-raw: bi-connector/schema-configuration -> ${base}/master/schema-configuration
-raw: bi-connector/installation -> ${base}/master/installation
-raw: bi-connector/components -> ${base}/master/components
-raw: bi-connector/components/mongobiuser -> ${base}/master/components
-raw: bi-connector/components/mongodrdl -> ${base}/master/components/mongodrdl
-raw: bi-connector/components/mongobischema -> ${base}/master/components
```
Comment (jeff-allen-mongo) [inline config/redirects:17]: [question]

Previously we had these two redirects:

1. `(v1.1-*]: bi-connector/${version}/reference/mongobischema -> ${base}/${version}/components`
2. `raw: bi-connector/components/mongobischema -> ${base}/master/components`

Now, we have:

` raw: bi-connector/components/mongobischema -> ${base}/master/components/`

Is that correct? Or should we retain two `mongobischema` redirects?

File: config/redirects [hunk 2]
```diff
@@ -1,43 +1,17 @@
...
-(v2.1-*]: bi-connector/${version}/tutorial/connecting -> ${base}/${version}/tutorial/using-the-bi
-(v2.1-*]: bi-connector/${version}/tutorial/connecting-to-atlas -> ${base}/${version}/reference/mongosqld
-[*-v2.1]: bi-connector/${version}/atlas-bi-connector/ -> ${base}/${version}
-[*-v2.1]: bi-connector/${version}/client-dependencies/ -> ${base}/${version}
-[*-v2.1]: bi-connector/${version}/tutorial/install-bi-connector-rhel/ -> ${base}/${version}/installation
-[*-v2.1]: bi-connector/${version}/tutorial/install-bi-connector-debian/ -> ${base}/${version}/installation
-[*-v2.1]: bi-connector/${version}/tutorial/install-bi-connector-macos/ -> ${base}/${version}/installation
-[*-v2.1]: bi-connector/${version}/tutorial/install-bi-connector-windows/ -> ${base}/${version}/installation
-[v2.0-v2.1]: bi-connector/${version}/client-applications/ -> ${base}/${version}/tutorial/using-the-bi
-(v2.1-*]: bi-connector/${version}/tutorial/using-the-bi/ -> ${base}/${version}/client-applications
```
Comment (jeff-allen-mongo) [inline config/redirects:31]: I don't see this redirect in the added ones. Should it be?

**Review by jeff-allen-mongo — COMMENTED** [review]
Thanks @schmalliso! I have two questions on this PR.

**Comment by schmalliso** [comment]
@jeff-allen-mongo  SO SORRY - I ended up fully revising this following a chat with Andrew A. about redirect deletion and such. I've reorganized and dealt with a few additional redirect chains, so let me know if you see anything! Sorry again!

File: config/redirects (hunk 2 above)
Comment (jeff-allen-mongo) [reply config/redirects:31]: Discussed via Slack. This redirect still needs to be added back.

File: config/redirects [hunk 3]
```diff
@@ -1,43 +1,43 @@
...
+[*-v2.1]: bi-connector/${version}/tutorial/install-bi-connector-macos/ -> ${base}/${version}/installation/
+[*-v2.1]: bi-connector/${version}/tutorial/install-bi-connector-windows/ -> ${base}/${version}/installation/
 [*-v2.1]: bi-connector/${version}/authentication/ -> ${base}/${version}/
+[*-v2.4]: bi-connector/${version}/reference/odbc-driver/ -> ${base}/${version}/
 [*-v2.7]: bi-connector/${version}/reference/system-variables/ -> ${base}/${version}/
+
+(v1.1-*]: bi-connector/${version}/reference/mongobiuser -> ${base}/${version}/components/
+(v1.1-*]: bi-connector/${version}/reference/mongobischema -> ${base}/${version}/components/
+(v2.1-*]: bi-connector/${version}/tutorial/connecting -> ${base}/${version}/client-applications/
+(v2.1-*]: bi-connector/${version}/tutorial/using-the-bi -> ${base}/${version}/client-applications/
```
Comment (schmalliso) [inline config/redirects:39]: @jeff-allen-mongo Added back in here!

File: config/redirects (hunk 2 above)
Comment (schmalliso) [reply config/redirects:31]: Added back in on line 39.

**Review by schmalliso — COMMENTED** [review]
@jeff-allen-mongo I've plopped it back in!

**Pull request merged**